valohai_yaml/objs/pipelines/override.py

```diff
@@ -69,3 +72,38 @@ def serialize(self) -> OrderedDict:  # type: ignore[type-arg]
                     elide_empty_iterables=True,
                 )
         return val
+
+    @classmethod
+    def default_merge(cls, a: Override | None, b: Any) -> Override:
```

Comment:
The name `default_merge` has a special meaning; it's supposed to be a merger for two objects of the same type.

If you don't mean to interfere with the `merge_with` mechanism here, maybe use another name (`apply_defaults_from_step` if that's what this is supposed to do?) and correct the names and types of the arguments too (and the locals – `result`?)